src/rand.rs

```diff
@@ -419,3 +422,14 @@ mod fuchsia {
         fn zx_cprng_draw(buffer: *mut u8, length: usize);
     }
 }
+
+#[cfg(any(target_os = "uefi"))]
+mod uefi {
+    use crate::error;
+    use rdrand::RdRand;
```

Comment:
Hi. Presently our policy is to not depend on third-party crates for any crypto primitives. Let's just use the `core::arch::x86_64` intrinsics directly, instead of depending on that crate.

At the top of this file is a long comment with a table that indicates the source of randomness for the target. Please add a comment about UEFI to that table.

Reply:
> Hi. Presently our policy is to not depend on third-party crates for any crypto primitives. Let's just use the core::arch::x86_64 intrinsics directly, instead of depending on that crate.

Scratch this. BoringSSL upstream has rdrand-x86_64.pl that exposes a `CRYPTO_rdrand` function that we can use instead. I also need to update cpu-intel.c to disable RDRAND on AMD CPUs due to bugs in some of them, and then I'll update cpu.rs so that you can detect whether RDRAND is available.